Description: Workaround for -fvisibility-inlines-hidden on alpha:
             break out the static methods so they're not inlined.
             fix FTBFS on alpha.
Bug-Debian: http://bugs.debian.org/368883
Author: Steve Langasek <vorlon@debian.org>

---
 src/corelib/global/qlibraryinfo.cpp |   14 ++++++++------
 src/corelib/tools/qhash.h           |    2 +-
 2 files changed, 9 insertions(+), 7 deletions(-)

--- a/src/corelib/global/qlibraryinfo.cpp
+++ b/src/corelib/global/qlibraryinfo.cpp
@@ -85,12 +85,7 @@ class QLibraryInfoPrivate
 {
 public:
     static QSettings *findConfiguration();
-    static void cleanup()
-    {
-        QLibrarySettings *ls = qt_library_settings();
-        if (ls)
-            ls->settings.reset(0);
-    }
+    static void cleanup();
     static QSettings *configuration()
     {
         QLibrarySettings *ls = qt_library_settings();
@@ -98,6 +93,13 @@ public:
     }
 };
 
+void QLibraryInfoPrivate::cleanup()
+{
+    QLibrarySettings *ls = qt_library_settings();
+    if (ls)
+        ls->settings.reset(0);
+}
+
 QLibrarySettings::QLibrarySettings()
     : settings(QLibraryInfoPrivate::findConfiguration())
 {
--- a/src/corelib/tools/qhash.h
+++ b/src/corelib/tools/qhash.h
@@ -523,7 +523,7 @@ Q_INLINE_TEMPLATE void QHash<Key, T>::de
 }
 
 template <class Key, class T>
-Q_INLINE_TEMPLATE void QHash<Key, T>::duplicateNode(QHashData::Node *node, void *newNode)
+void QHash<Key, T>::duplicateNode(QHashData::Node *node, void *newNode)
 {
     Node *concreteNode = concrete(node);
     if (QTypeInfo<T>::isDummy) {
